waterdata/templates/monitoring_location.html

```diff
@@ -31,6 +30,8 @@
 
 {% set body_id = 'monitoring-location' %}
 
+{% if stations|length == 1 %}{% set page_title = stations[0].station_nm %}{% endif %}
```

Comment:
Rather than putting it here, put this line after the import for the macro/components. The reason it didn't work where you put it is that it is only defined within the {% block extra_head_tags %}. By putting it outside of a block it will be available anywhere in the template.